cobalt/worker/service_worker_jobs.cc

```diff
@@ -40,6 +41,14 @@ bool PathContainsEscapedSlash(const GURL& url) {
 }
 
 bool IsOriginPotentiallyTrustworthy(const GURL& url) {
+#if !defined(COBALT_BUILD_TYPE_GOLD)
```

Comment:
The pre-existing code for this for web platform tests is the `(origin.host() == "web-platform.test")` below on line 87. 

The Chrome code has an allowlist that can be appended from the command line (cmdline_allowlist_), and a command line switch `kUnsafelyTreatInsecureOriginAsSecure` (`unsafely-treat-insecure-origin-as-secure`).

I think it would be good if we can mirror Chrome's approach here.

https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/cpp/is_potentially_trustworthy.cc;l=328;drc=618111ac68d8ed2e3a84eee9cae1cf61ce9ba236

https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/cpp/is_potentially_trustworthy.cc;l=419;drc=d4ab3a0ab41ca38347f06d8a6282cf2fa4bfaed8

https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/cpp/is_potentially_trustworthy.cc;l=228;drc=d4ab3a0ab41ca38347f06d8a6282cf2fa4bfaed8

https://source.chromium.org/chromium/chromium/src/+/main:services/network/public/cpp/network_switches.cc;l=57-65;drc=18f229840b08685f49b7f0f514a3597cd856618e